### PR TITLE
Added source url to search indexes and missing entites

### PR DESCRIPTION
**Pull request by 07Himank**
Fixes : #11178

I worked on ... because ... added SourceUrl to the missing entites as well as search index mapping files

#
### Type of change:

- [ x] Bug fix
- [ x] Improvement

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/api/data/createTopic.json [hunk 1]
```diff
@@ -80,6 +80,10 @@
       "description": "Entity extension data with custom attributes added to the entity.",
       "$ref": "../../type/basic.json#/definitions/entityExtension"
     },
+    "sourceUrl": {
+      "description": "Source URL of topic.",
+      "$ref": "../../type/basic.json#/definitions/sourceUrl"
```
Comment (harshach) [inline openmetadata-spec/src/main/resources/json/schema/api/data/createTopic.json:85]: @07Himank we already have URL type, take a look at that

**Comment by harshach** [comment]
@07Himank lets create a separate api in search resource to be able to look up by just passing URL without adding any other ranking etc..

**Comment by 07Himank** [comment]
> @07Himank lets create a separate api in search resource to be able to look up by just passing URL without adding any other ranking etc..

@harshach , Thats what i wanted to confirm we need to add a separate API to get entities from sourceURL (with exact match ) right ? as we discussed

File: openmetadata-spec/src/main/resources/json/schema/api/data/createTopic.json (hunk 1 above)
Comment (07Himank) [reply openmetadata-spec/src/main/resources/json/schema/api/data/createTopic.json:85]: @harshach I took reference from dashboard.json , where sourceURL was already there do we need to update that ?

File: openmetadata-service/src/main/java/org/openmetadata/service/search/elasticSearch/ElasticSearchClientImpl.java [hunk 2]
```diff
@@ -363,6 +363,29 @@ public Response search(ElasticSearchRequest request) throws IOException {
...
 
+  /**
+   * @param sourceUrl
+   * @return
+   */
+  @Override
+  public Response searchBySourceUrl(String sourceUrl) throws IOException {
+    String[] indexes =
+        new String[] {
+          ElasticSearchIndexDefinition.ElasticSearchIndexType.TABLE_SEARCH_INDEX.indexName,
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/search/elasticSearch/ElasticSearchClientImpl.java:374]: we should create alias for the indexes and search across all the indexes at once using the alias

Comment (07Himank) [reply openmetadata-service/src/main/java/org/openmetadata/service/search/elasticSearch/ElasticSearchClientImpl.java:374]: thanks for review, will push the changes..

**Review by mohityadav766 — APPROVED** [review]

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**